iceoryx_posh/doc/chunk_header.md

```diff
@@ -0,0 +1,218 @@
+
+3. Custom header is used
+
+```
+  sizeof(ChunkHeader)            back-offset    m_payloadSize
+|-------------------->|                |<---|------------------->|
+|                     |                |    |                    |
++=====================+=====================+====================+============+
+|                     |   Custom    |  .    |                    |            |
+|     ChunkHeader     | ChunkHeader |  .    |      Payload       |  Padding   |
+|                     |  Extension  |  .    |                    |            |
++=====================+=====================+====================+============+
+|                                           |                                 |
+|           m_payloadOffset                 |                                 |
+|------------------------------------------>|                                 |
+|                                                                             |
+|                                m_chunkSize                                  |
+|---------------------------------------------------------------------------->|
+```
+
+### Payload Offset Calculation
+
+1. No custom header and payload alignment doesn't exceed the `ChunkHeader` alignment
+
+```
+payloadOffset = sizeof(ChunkHeader);
+```
+
+2. No custom header and payload alignment exceeds the `ChunkHeader` alignment, which means the payload is either adjacent to the `ChunkHeader` or there is a padding with at least the size of `ChunkHeader` in front of the payload and therefore enough space to store the `back-offset`
+
+```
+headerEndAddress = addressof(chunkHeader) + sizeof(chunkHeader);
+payloadAddress = align(headerEndAddress, payloadAlignment);
+payloadOffset = payloadAddress - addressof(chunkHeader);
+```
+
+3. Custom header is used
+
+```
+headerEndAddress = addressof(chunkHeader) + sizeof(chunkHeader) + sizeof(customHeader);
+potentialBackOffsetAddress = align(headerEndAddress, alignof(payloadOffset));
+potentialPayloadAddress = potentialBackOffsetAddress + sizeof(payloadOffset);
+payloadAddress = align(potentialPayloadAddress, payloadAlignment);
+payloadOffset = payloadAddress - addressof(chunkHeader);
+```
+
+### Required Chunk Size Calculation
+
+In order to fit the custom header and the payload into he chunk, a worst case calculation has to be done. We can assume that a chunk is aligned to 32 bytes, which is also the alignment of the `ChunkHeader`.
+
+1. No custom header and payload alignment doesn't exceed the `ChunkHeader` alignment
+
+```
+chunkSize = sizeof(chunkHeader) + payloadSize;
+```
+
+2. No custom header and payload alignment exceeds the `ChunkHeader` alignment
+
+Worst case scenario is when a part of the `ChunkHeader` crosses the payload alignment boundary, so that the payload must be aligned to the next boundary.
+```
```

Comment:
yes, you are right, as long as the size of `ChunkHeader` does not exceed the alignment, it's 0. This makes it just future prove in case this changes, e.g. because we decide to decrease the alignment because we would waste memory when the actual size would be e.g. 40 bytes. The chunks could still be aligned to 32 bytes, but the `ChunkHeader` alignment could be 8.

Reply:
updated with some ASCII art